de.metas.business/src/main/java/org/adempiere/impexp/product/MProductImportTableSqlUpdater.java

```diff
@@ -446,6 +452,19 @@ private void dbUpdateDosageForm(@NonNull final String whereClause)
 		DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
 	}
 
+	private void dbUpdatePharmaProductCategory(@NonNull final String whereClause)
+	{
+
+		final StringBuilder sql = new StringBuilder("UPDATE ")
+				.append(targetTableName + " i ")
+				.append(" SET M_PharmaProductCategory_ID=(SELECT M_PharmaProductCategory_ID FROM M_PharmaProductCategory c")
+				.append(" WHERE i.PharmaProductCategory_Value=c.Name AND i.AD_Client_ID=c.AD_Client_ID) ")
+				.append("WHERE PharmaProductCategory_Value IS NOT NULL AND M_PharmaProductCategory_ID IS NULL")
+				.append(" AND " + COLUMNNAME_I_IsImported + "<>'Y'").append(whereClause);
+
+		DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
```

Comment:
* why is it `I_Product.PharmaProductCategory_Value`, but `M_PharmaProductCategory.Name`?
  * pls sync them
* is `M_PharmaProductCategory.Name` read-only? if not, i guess it would be goot to make it read-only

Reply:
Yes, good point! Thx!